interview-questions.md

```diff
@@ -8,44 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 2. What is a gem?
 
-Your answer:
+Your answer: A gem is a package manager for Ruby. It works similar to yarn for JavaScript. Gems can be thought of as methods. These methods can be utilized in other programs, or "objects", that will allow the user to perform certain functions without actually inserting the code. Gems are used for package organization, package modification and making shortcuts for certain processes.
 
-Researched answer:
+Researched answer: Gems are essentially a library of rules that shape the way a Ruby application works. 
 
 3. What is Ruby on Rails?
 
-Your answer:
+Your answer: Rails is a framework written in the ruby language. It is similar to React for Javascript. Rails basically provides a background structure for code to run on. This background structure has rules programmed into it that allow Ruby code to interact a certain way according to certain conditions. 
 
-Researched answer:
+Researched answer: Rails is commonly used to create web applications. 
 
 4. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A database of elements that share a specific relationship. These relationships, within the database, can shape the database. As a result, the database is compiled of elements that are organized based on how they relate to one another. 
 
-Researched answer:
+Researched answer: Relational databases are useful for data that would be organized in the form of a table. Other kinds of databases include Hierarchical, Network, Object-oriented and NoSQL. 
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: Primary keys are identifiers for rows inside database tables. The have a similar representation to keys for key value pairs. Primary keys are necessary for referencing rows of data for organization and modification.  
 
-Researched answer:
+Researched answer: Primary keys are the difference between locating specific data and not. In a table that is abundant with data, without an identifier, it would tedious to locate specifc data for a given data table. Primary keys provide a shortcut to pinpointing a location in a given data table.
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
-1. RESTful routes:
+1. RESTful routes: Representational State Transfer routes. Connect HTTP methods to Crud actions in order to create default pattern for certain behaviors.
 
-2. JSON:
+2. JSON: JavaScript Object Notation. A data interchange format that uses text to store and transmit data. 
 
-3. ERB:
+3. ERB: Embredded Ruby. Allows Ruby code to be transitioned to text. 
 
-4. Params:
+4. Params: A keyword that allows user to pass a variable number of parameters into a method. 
 
-5. API:
+5. API: Application Programming Interface. A software interface that allows multiple programs to communicate with each other. 
```

Comment:
Excellent work on your own answers.  You may want to get more information when you research the question.  The more info you can learn, understand, and communicate, the better you will do in interviews.